src/screens/MyWorkouts/MyWorkouts.tsx

```diff
@@ -25,11 +29,28 @@ const MyWorkouts: React.FC<MyWorkoutsProps> = ({ navigation }) => {
   };
 
   const updateWorkout = async (payload) => {
-    await updateEditableWorkout(payload);
+    await updateEditableWorkout(payload.id, payload);
     setEditingWorkout(undefined);
     if (refreshWorkouts) await refreshWorkouts();
   };
-
+  const deleteWorkoutBlock = async (id) => {
+    await deleteEditableWorkout(id);
+    setEditingWorkout(undefined);
+    if (refreshWorkouts) await refreshWorkouts();
+  };
+  const duplicateWorkoutBlock = async (payload: EditableWorkout) => {
+    const workout = await createEditableWorkout(session);
+    if (workout) {
+      setEditingWorkout(workout.id);
```

Comment:
You don't need to set this, it's really just for the renaming. That's my bad though because its confusing naming.